test/lib/test-ringpop-cluster.js

```diff
@@ -127,12 +126,66 @@ function testRingpopCluster(opts, name, test) {
     }
 
     tape(name, function onTest(assert) {
-        var cluster = bootstrapClusterOf(opts, function onBootstrap(results) {
-            assert.on('end', function onEnd() {
-                destroyCluster(cluster);
+        var tmpMinProtocolPeriods = [];
+        var cluster = createClusterOf(opts);
+        var joinResults;
+
+        // store protocol period. Set protocol period to 1 so that convergence is fast
+        cluster.forEach(function each(ringpop, i) {
+            tmpMinProtocolPeriods[i] = ringpop.gossip.minProtocolPeriod;
+            ringpop.gossip.minProtocolPeriod = 1;
+        });
+
+        function onConverged() {
+            // do not run onConverged if not all joins are succesful
+            if (joinResults === undefined) {
+                return;
+            }
+
+            // revert back to original protocol period
+            cluster.forEach(function each(ringpop, i) {
+                ringpop.gossip.minProtocolPeriod = tmpMinProtocolPeriods[i];
             });
 
-            test(results, cluster, assert);
+            if (typeof opts.tapAfterConvergence === 'function') {
+                opts.tapAfterConvergence(cluster);
+            }
+
+            if (opts.waitForConvergence !== false && opts.checkChecksums === true) {
+                var checksums = _.chain(cluster)
+                    .pluck('membership')
+                    .pluck('checksum')
+                    .uniq();
+                if (!(checksums.length === 1 && typeof checksums[0] === 'number')) {
+                    assert.fail('not all checksums are equal');
+                }
+            }
+
+            // cluster.forEach(function each(ringpop) {
+            //     ringpop.minProtocolPeriod = minProtocolPeriod;
+            // });
+
+            test(joinResults, cluster, assert);
+        }
+
+        var nodeExhausted = _.after(cluster.length, onConverged);
+        
+        cluster.forEach(function each(ringpop) {
+            ringpop.dissemination.on('changesExhausted', nodeExhausted);
```

Comment:
can you rename event handler to `onExhausted`? That's a convention that screams out "callback"